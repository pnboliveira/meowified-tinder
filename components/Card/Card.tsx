import React from "react";
import { Text, ImageBackground, StyleSheet, View } from "react-native";

// create Card props with their types
interface CardProps {
  name?: string;
  bio?: string;
  image: string;
  adaptability?: number;
}

const Card = ({ name, bio, image, adaptability }: CardProps) => {
  return (
    <View style={styles.card}>
      <ImageBackground
        source={{
          uri: image,
        }}
        style={styles.image}
      >
        <View style={styles.cardInner}>
          <Text style={styles.name}>{name}</Text>
          <Text style={styles.adaptability}>Adaptability: {adaptability}</Text>
          <Text style={styles.bio}>{bio}</Text>
        </View>
      </ImageBackground>
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    width: "95%",
    height: "70%",
    borderRadius: 10,
    backgroundColor: "#FFF",
  },
  image: {
    width: "100%",
    height: "100%",
    borderRadius: 10,
    overflow: "hidden",
    justifyContent: "flex-end",
  },
  cardInner: {
    alignSelf: "center",
    justifyContent: "space-evenly",
    width: 350,
    height: 60,
    top: 8,
    paddingTop: 8,
    paddingLeft: 16,
    backgroundColor: "#FFF",
    borderRadius: 16,
    gap: 2,
  },
  name: {
    fontSize: 16,
    lineHeight: 20,
    fontWeight: "bold",
  },
  bio: {
    lineHeight: 10,
    fontSize: 8,
  },
});

export default Card;
