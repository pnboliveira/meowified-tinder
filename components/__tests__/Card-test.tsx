import * as React from "react";
import renderer from "react-test-renderer";

import Card from "@/components/Card/Card";

it(`renders correctly`, () => {
  const tree = renderer
    .create(
      <Card
        name="John Doe"
        bio="I am a bio"
        image="https://example.com/image.png"
        adaptability={7}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
