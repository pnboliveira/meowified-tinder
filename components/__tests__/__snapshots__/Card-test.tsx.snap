// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "borderRadius": 10,
      "height": "70%",
      "width": "95%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "borderRadius": 10,
        "height": "100%",
        "justifyContent": "flex-end",
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://example.com/image.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 16,
          "gap": 2,
          "height": 60,
          "justifyContent": "space-evenly",
          "paddingLeft": 16,
          "paddingTop": 8,
          "top": 8,
          "width": 350,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 20,
          }
        }
      >
        John Doe
      </Text>
      <Text>
        Adaptability: 
        7
      </Text>
      <Text
        style={
          {
            "fontSize": 8,
            "lineHeight": 10,
          }
        }
      >
        I am a bio
      </Text>
    </View>
  </View>
</View>
`;
